doc/source/serve/advanced-guides/grpc-guide.md

```diff
@@ -290,3 +291,32 @@ Serve uses the following gRPC error codes:
   this node and you should no longer route to this node.
 - `DEADLINE_EXCEEDED`: The request took longer than the timeout setting and got cancelled.
 - `INTERNAL`: Other unhandled errors during the request.
+
+(serve-grpc-proxy-grpc-context)=
+## Use gRPC context
+Serve provides a gRPC context object to the deployment replica to get information
+about the request as well as setting response metadata such as code and details.
+Once the deployment is defined with a `grpc_context` argument, Serve will pass a
+[RayServegRPCContext](../api/doc/ray.serve.grpc_util.RayServegRPCContext.rst) object
+to the replica. Let's define the deployment like the following to set custom code,
+details, and trailing metadata.
+
+```{literalinclude} ../doc_code/grpc_proxy/grpc_guide.py
+:start-after: __begin_grpc_context_define_app__
+:end-before: __end_grpc_context_define_app__
+:language: python
+```
+
+We can define the client code like the following to get the custom code, details, and
+trailing metadata.
```

Comment:
This is redundant with the above. Maybe just remove it, or else "refactor" and remove the last sentence from the above paragraph and only have the information here.